tqdm/_tqdm.py

```diff
@@ -734,6 +752,10 @@ def __iter__(self):
                                 else smoothing * delta_t / delta_it + \
                                 (1 - smoothing) * avg_time
 
+                        # Acquire locks if parallel bars
```

Comment:
So just find a purpose to make a program of your own, and continue. The goal is not to learn to program, but simply to achieve a program that works for your needs. In the end, you will progress a lot. And this works at any proficiency level, I am still learning everyday in the languages I master the most.